content/articles/How-to-start-up-with-Socket-Programming-with-Java/index.md

```diff
@@ -0,0 +1,331 @@
+2. `public OutputStream( getOutputStream())`- returns an OutputStream for sending data.
+3. `BufferReader( )` - Wraps inputStreamReader to improve efficiency
+4. `BufferWriter( )` - Wraps outputStreamReader to improve efficiency
+5. `BufferedWriter.flush( )` - Flushes the streams...The flush method is invoked when the buffer is complete.
+6.  `void close() `- closes all streams and the socket.
+  
+### THIS IS A CLIENT PROGRAM THAT WILL BE COMMUNICATING WITH THE SERVER.
+```java
+package com.company;  
+  
+import java.io.*;  
+import java.net.Socket;  
+import java.util.Scanner;  
+  
+public class client {  
+ public static void main(String[] args) {  
+
+  Socket socket = null;  
+  InputStreamReader inputStreamReader = null;  
+  OutputStreamWriter outputStreamWriter = null;  
+  BufferedReader bufferedReader = null;  
+  BufferedWriter bufferedWriter = null;  
+  
+try {  
+socket = new Socket("localhost", 5000);  
+inputStreamReader = new InputStreamReader(socket.getInputStream());  
+outputStreamWriter = new OutputStreamWriter(socket.getOutputStream());  
+
+bufferedReader = new BufferedReader(inputStreamReader);  
+bufferedWriter = new BufferedWriter(outputStreamWriter);  
+
+Scanner scanner = new Scanner(System.in);  
+
+while (true)
+ {  
+  String msgToSend = scanner.nextLine();  
+
+  bufferedWriter.write(msgToSend);  
+
+  bufferedWriter.newLine();  
+
+  bufferedWriter.flush();  
+  
+  System.out.println("Server: " + bufferedReader.readLine());  
+  
+if (msgToSend.equalsIgnoreCase("BYE"))  
+    break;  
+  }  
+} catch (IOException e) {  
+   e.printStackTrace();  
+} finally {  
+try {  
+
+if (socket != null)  
+ socket.close();  
+if (inputStreamReader != null)  
+  inputStreamReader.close();  
+if (outputStreamWriter != null)  
+ outputStreamWriter.close();  
+if (bufferedReader != null)  
+ bufferedReader.close();  
+if (bufferedWriter != null)  
+ bufferedWriter.close();  
+} catch (IOException e) {  
+  e.printStackTrace();  
+          }  
+        }  
+    }  
+}
+```
+The output of the code:
+
+```
+Hey there 
+Server:  Message Received
+am doing this for the first time
+Server:  Message Received
+BYE
+Server:  Message Received
+
+```
+### Procedure of How to run the two programs if using Intellij or any other IDE
+1. Compile the two 
+2. First execute the program
+3. on the server **run**, then run the program client **run* 
+4. Type client window messages that are simultaneously received and shown by the Server window.
+5. Type **BYE** to end.
+
+### If using Command prompt/Terminal
+1. Depending on how you stored it, start the server program as the java server.
+2. Then there are two. Run the client application on a terminal like a java customer.
+3. Then Type messages in the client window
+
+Example 
+```
+Hey there 
+Server:  Message Received
+am doing this for the first time
+Server:  Message Received
+BYE
+Server:  Message Received
+```
+4.The server receives and simultaneously shows;
+
+```bash
```

Comment:
Don't label this output snippet as *bash*, it may try to add code highlighting to certain words when you are just writing in plain text.

Change: 

>```bash

to: *```*